src/Compilers/CSharp/Test/Semantic/Semantics/AmbiguousOverrideTests.cs

```diff
@@ -212,28 +203,92 @@ public void TestAmbiguousOverridesRefOut()
+    public virtual void Method(out List<A> x)
+    {
+        x = null;
+        Console.WriteLine(""Base2<A, B>.Method(out List<A> x)"");
+    }
 }
 class Derived : Base2<int, int>
 {
-    public override void Method(ref List<int> a, out List<int> b) { b = null; } // Reports warning about runtime ambiguity
-    public override void Method(ref List<int> a) { } // No warning when ambiguous signatures are spread across multiple base types
-}";
-
-            CreateCompilation(source).VerifyDiagnostics(
-                // (5,25): warning CS1957: Member 'Derived.Method(ref System.Collections.Generic.List<int>, out System.Collections.Generic.List<int>)' overrides 'Base<int, int>.Method(ref System.Collections.Generic.List<int>, out System.Collections.Generic.List<int>)'. There are multiple override candidates at run-time. It is implementation dependent which method will be called.
-                Diagnostic(ErrorCode.WRN_MultipleRuntimeOverrideMatches, "Method").WithArguments("Base<int, int>.Method(ref System.Collections.Generic.List<int>, out System.Collections.Generic.List<int>)", "Derived.Method(ref System.Collections.Generic.List<int>, out System.Collections.Generic.List<int>)"));
+    public override void Method(ref List<int> a, out List<int> b) // No warning - the compiler produces a methodimpl record to disambiguate
+    {
+        b = null;
+        Console.WriteLine(""Derived.Method(ref List<int> a, out List<int> b)"");
+    }
+    public override void Method(ref List<int> a) // No warning when ambiguous signatures are spread across multiple base types
+    {
+        Console.WriteLine(""Derived.Method(ref List<int> a)"");
+    }
+}
+class Program
+{
+    static void Main()
+    {
+        List<int> t = null;
+        Derived d = new Derived();
+        d.Method(ref t, out t);
+        d.Method(out t, ref t);
+        d.Method(ref t);
+        d.Method(out t);
+        Console.WriteLine();
+        Base2<int, int> b2 = d;
+        b2.Method(ref t, out t);
+        b2.Method(out t, ref t);
+        b2.Method(ref t);
+        b2.Method(out t);
+        Console.WriteLine();
+        Base<int, int> b1 = d;
+        b1.Method(ref t, out t);
+        b1.Method(out t, ref t);
+        b1.Method(ref t);
+    }
+}
+";
+            var expectedOutput = @"
+Derived.Method(ref List<int> a, out List<int> b)
+Base<T, U>.Method(out List<U> y, ref List<T> x)
+Derived.Method(ref List<int> a)
+Base2<A, B>.Method(out List<A> x)
+
+Derived.Method(ref List<int> a, out List<int> b)
+Base<T, U>.Method(out List<U> y, ref List<T> x)
+Derived.Method(ref List<int> a)
+Base2<A, B>.Method(out List<A> x)
+
+Derived.Method(ref List<int> a, out List<int> b)
+Base<T, U>.Method(out List<U> y, ref List<T> x)
+Derived.Method(ref List<int> a)";
+            for (int i = 0; i < 2; i++)
+            {
+                if (i == 1)
+                {
+                    // This case is not working due to https://github.com/dotnet/runtime/issues/38119
```

Comment:
>// This case is not working due to https://github.com/dotnet/runtime/issues/38119 [](start = 20, length = 81)

Based on that please revert removal of the ambiguity warning. We can revisit its removal once, if runtime behavior is fixed. Which likely never for desktop.

Reply:
The semantics of compilation is targeting core clr. The runtime will have to be fixed for the covariant return feature to work properly.

---
In reply to: [442540260](https://github.com/dotnet/roslyn/pull/44025#discussion_r442540260) [](ancestors = 442540260)

Reply:
This is being tracked by [https://github.com/dotnet/roslyn/issues/45453](https://github.com/dotnet/roslyn/issues/45453)

---
In reply to: [442546841](https://github.com/dotnet/roslyn/pull/44025#discussion_r442546841) [](ancestors = 442546841,442540260)